### ISSUE TITLE
Zero-dimensional scalars

### ISSUE THREAD

**Issue by TomNicholas** [TomNicholas/VirtualiZarr#100]
Turns out there is a bug if you try to open a zero-dimensional scalar:

```python
In [6]: import xarray as xr

In [7]: ds = xr.Dataset(data_vars={"a": 0})

In [8]: ds.to_netcdf('scalar.nc')

In [9]: ds = open_virtual_dataset("scalar.nc")
a
{'.zarray': {'chunks': [], 'compressor': None, 'dtype': '<i8', 'fill_value': -9223372036854775806, 'filters': None, 'order': 'C', 'shape': [], 'zarr_format': 2}, '.zattrs': {'_ARRAY_DIMENSIONS': []}, '0': ['scalar.nc', 590, 8]}
chunks=() compressor=None dtype=dtype('int64') fill_value=-9.223372036854776e+18 filters=None order='C' shape=() zarr_format=2
entries={'0': ChunkEntry(path='scalar.nc', offset=590, length=8)}
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[9], line 1
----> 1 ds = open_virtual_dataset("scalar.nc")

File ~/Documents/Work/Code/virtualizarr/virtualizarr/xarray.py:95, in open_virtual_dataset(filepath, filetype, drop_variables, loadable_variables, indexes, virtual_array_class)
     88 else:
     89     # this is the only place we actually always need to use kerchunk directly
     90     # TODO avoid even reading byte ranges for variables that will be dropped later anyway?
     91     vds_refs = kerchunk.read_kerchunk_references_from_file(
     92         filepath=filepath,
     93         filetype=filetype,
     94     )
---> 95     virtual_vars = virtual_vars_from_kerchunk_refs(
     96         vds_refs,
     97         drop_variables=drop_variables + loadable_variables,
     98         virtual_array_class=virtual_array_class,
     99     )
    100     ds_attrs = kerchunk.fully_decode_arr_refs(vds_refs["refs"]).get(".zattrs", {})
    102     if indexes is None or len(loadable_variables) > 0:
    103         # TODO we are reading a bunch of stuff we know we won't need here, e.g. all of the data variables...
    104         # TODO it would also be nice if we could somehow consolidate this with the reading of the kerchunk references
    105         # TODO really we probably want a dedicated xarray backend that iterates over all variables only once

File ~/Documents/Work/Code/virtualizarr/virtualizarr/xarray.py:215, in virtual_vars_from_kerchunk_refs(refs, drop_variables, virtual_array_class)
    210     drop_variables = []
    211 var_names_to_keep = [
    212     var_name for var_name in var_names if var_name not in drop_variables
    213 ]
--> 215 vars = {var_name: variable_from_kerchunk_refs(
    216         refs, var_name, virtual_array_class
    217     ) for var_name in var_names_to_keep}
    219 return vars

File ~/Documents/Work/Code/virtualizarr/virtualizarr/xarray.py:274, in variable_from_kerchunk_refs(refs, var_name, virtual_array_class)
    271 print(zarray)
    272 print(manifest)
--> 274 varr = virtual_array_class(zarray=zarray, chunkmanifest=manifest)
    276 return xr.Variable(data=varr, dims=dims, attrs=zattrs)

File ~/Documents/Work/Code/virtualizarr/virtualizarr/manifests/array.py:59, in ManifestArray.__init__(self, zarray, chunkmanifest)
     57 # Check that the chunk grid implied by zarray info is consistent with shape implied by chunk keys in manifest
     58 if _zarray.shape_chunk_grid != _chunkmanifest.shape_chunk_grid:
---> 59     raise ValueError(
     60         f"Inconsistent chunk grid shape between zarray info and manifest: {_zarray.shape_chunk_grid} vs {_chunkmanifest.shape_chunk_grid}"
     61     )
     63 self._zarray = _zarray
     64 self._manifest = _chunkmanifest

ValueError: Inconsistent chunk grid shape between zarray info and manifest: () vs (1,
```

**Comment by TomNicholas** [comment]
This issue made me realise that in the zarr model knowing the array shape & chunk shape is not actually enough information to deduce the shape of the chunk grid, because there isn't a one-to-one mapping between them.

For example, an array with a single chunk `0` might be associated with the shape `()`, or `(X,)`.

**Comment by TomNicholas** [comment]
> EDIT: There is probably some more work to be done here with making sure expand_dims works for scalars.

This should be fixed by #103
